src/main/java/com/aws/greengrass/util/platforms/unix/linux/CGroupV2.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Set;
+
+@SuppressFBWarnings(value = "DMI_HARDCODED_ABSOLUTE_FILENAME",
+        justification = "CGroupV2 virtual filesystem path cannot be relative")
+public enum CGroupV2 implements CGroupSubSystemPaths {
+    Memory, CPU, Freezer, Unified;
+    private static final String CGROUP_SUBTREE_CONTROL_CONTENT = "+cpuset +cpu +io +memory +pids";
+
+    @Override
+    public String rootMountCmd() {
+        return String.format("mount -t cgroup2 none %s", CGROUP_ROOT);
+    }
+
+    @Override
+    public Path getSubsystemRootPath() {
+        return CGROUP_ROOT;
+    }
+
+    @Override
+    public Path getComponentMemoryLimitPath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(MEMORY_MAX);
+    }
+
+    @Override
+    public Path getCgroupFreezerStateFilePath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(CGROUP_FREEZE);
+    }
+
+    @Override
+    public Path getRootSubTreeControlPath() {
+        return getSubsystemRootPath().resolve(CGROUP_SUBTREE_CONTROL);
+    }
+
+    @Override
+    public Path getGGSubTreeControlPath() {
+        return getSubsystemGGPath().resolve(CGROUP_SUBTREE_CONTROL);
+    }
+
+    @Override
+    public Path getComponentCpuMaxPath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(CPU_MAX);
+    }
+
+    @Override
+    public Path getCgroupFreezePath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(CGROUP_FREEZE);
+    }
+
+    @Override
+    public void initializeCgroup(GreengrassService component, LinuxPlatform platform) throws IOException {
+        Set<String> mounts = getMountedPaths();
+
+        if (!mounts.contains(getRootPath().toString())) {
+            platform.runCmd(rootMountCmd(), o -> {
+            }, "Failed to mount cgroup root");
+            Files.createDirectory(getSubsystemRootPath());
+        }
+
+        if (!Files.exists(getSubsystemGGPath())) {
+            Files.createDirectory(getSubsystemGGPath());
+        }
+        if (!Files.exists(getSubsystemComponentPath(component.getServiceName()))) {
+            Files.createDirectory(getSubsystemComponentPath(component.getServiceName()));
+        }
+
+        //Enable controllers for root group
+        Files.write(getRootSubTreeControlPath(),
+                CGROUP_SUBTREE_CONTROL_CONTENT.getBytes(StandardCharsets.UTF_8));
+        //Enable controllers for gg group
+        Files.write(getGGSubTreeControlPath(),
+                CGROUP_SUBTREE_CONTROL_CONTENT.getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Override
+    public void handleCpuLimits(GreengrassService component, double cpu) throws IOException {
```

Comment:
Same question here, this looks mostly the same as the V1 implementation, can they share the same implementation?

Reply:
Method handleCpuLimits use different file paths and data format in the  V1 and V2. I have no idea how to make them into one method. Do you have any suggestion?